upt_macports/upt_macports.py

```diff
@@ -51,6 +51,26 @@ def run_depends(self):
     def test_depends(self):
         return self._depends('test')
 
+    @property
+    def archive_type(self):
+        archive_keyword = {
+            'gz': 'gz',
+            '7z': '7z',
+            'bz2': 'bzip2',
+            'lzma': 'lzma',
+            'tar': 'tar',
+            'zip': 'zip',
+            'xz': 'xz'
+        }
+        try:
+            archive_name = self.upt_pkg.get_archive().filename
+            archive_type = archive_name.split('.')[-1]
+            return archive_keyword.get(archive_type, 'unknown')
+
+        except upt.ArchiveUnavailable:
+            self.logger.error(f'Detection of archive type failed: no archive found')
```

Comment:
Do not use an f-string when it is not needed. I also think this should rather read something like "Could not determine the type of the source archive". Writing "no archive found" makes me think upt-macports was looking for the archive, while it was upt-pypi which failed to find one.

Reply:
> Also, if the line is too long, you can add "# noqa" at the end to disable the linter for this particular line.

Oo I had no idea about this, I thought global test settings bust be changed.

I have done the new changes as suggested.